code/modules/mob/living/silicon/robot/robot_modules.dm

```diff
@@ -484,13 +487,48 @@
 	moduleselect_icon = "service"
 	special_light_key = "service"
 	hat_offset = 0
+	whitelisted_reagents += list("orangejuice" , "tomatojuice") //Add here anything it can synthetize regardless of emag status
 
 /obj/item/robot_module/butler/respawn_consumable(mob/living/silicon/robot/R, coeff = 1)
 	..()
 	var/obj/item/reagent_containers/O = locate(/obj/item/reagent_containers/food/condiment/enzyme) in basic_modules
 	if(O)
 		O.reagents.add_reagent("enzyme", 2 * coeff)
 
+/obj/item/robot_module/proc/analyze_reagents(var/list/reagents_inside)
+	var/mob/living/silicon/robot/R = loc
+	to_chat(loc, "<span class='notice'>You start analyzing the solution.</span>")
+	var/i = 0
+	for(var/each_reagent in reagents_inside) //If there are multiple reagents
+		i += 10 //So it kinda prevents spam
+		if(each_reagent in can_synthetize) //If you can already synthetize it, no need to analyze again
+			addtimer(to_chat(loc, "<span class='notice'>You already successefully analyzed [each_reagent].</span>"), i)
+		else
+			addtimer(to_chat(loc, "<span class='notice'>Analyzing unknown reagent.</span>"), i)
+			R.cell.use(15) //Consumes energy to analyze. Not sure what happens if the borg do not have enough energy. Will consume energy even if it fails to analyze
+			addtimer(to_chat(loc, "<span class='notice'>You successefully analyzed [each_reagent]!.</span>"), 300 + i) //Add a timer to simulate analyzing
+			if(each_reagent in blacklisted_reagents)

```

Comment:
So, the can_synth is to prevent the reagent to be analyzed again. If can_synth, I can make it and do not need to analyze. If !can_synth it means it needs to be analyzed to see if it will go to the can_synth list

First step: Checks if I already know the reagent(If it exists in list\can_synth). If I dont, moving on
Second step: Checks if the reagent is blacklisted. If it is, just add a delay because it is being analyzed
Third step: Checks if either the reagent is whitelisted or the borg is emagged (which in this case adds to the can_synth list)
Fourth step: The reagent is not blacklisted but is also not whitelisted, it doesn't go to the can_synth.

This interaction creates the need for the bot to first "research" the reagent to be able to synthetize it.

Reply:
`can_synth` is a reagent var that determines if the chem is globally blacklisted for replication, it's used for stuff like the odysseus and random reagents.

Reply:
I was literally confused because I made a list called "can_synthesize" and when you said about "can_synth" I thought it was the same thing. To avoid confusion, will change the name to something else, like "know_how_to_synth" or something like that

Reply:
Removed the blacklist thing and added the can_synth check. Also renamed the var to avoid confusion in case someone mess with this in the future